packages/astro/src/runtime/client/dev-toolbar/apps/audit/perf.ts

```diff
@@ -0,0 +1,126 @@
+import type { AuditRuleWithSelector } from './index.js';
+
+// A regular expression to match external URLs
+const externalUrlRe = /^(?:[a-z+]+:)?\/\//i;
+
+export const perf: AuditRuleWithSelector[] = [
+	{
+		code: 'perf-use-image-component',
+		title: 'Use the Image component',
+		message: 'This image could be replaced with the Image component to improve performance.',
+		selector: 'img:not([data-image-component])',
+		async match(element) {
+			const src = element.getAttribute('src');
+			if (!src) return false;
+
+			// Don't match data URIs, they're typically used for specific use-cases that the image component doesn't help with
+			if (src.startsWith('data:')) return false;
+
+			// Ignore images that are smaller than 20KB, most of the time the image component won't really help with these, or they're used for specific use-cases (pixel tracking, etc.)
+			// Ignore this test for remote images for now, fetching them can be very slow and possibly dangerous
```

Comment:
Will we document these exceptions somewhere?

Reply:
Eventually, yes! We're planning to have a list of audit page on the docs, like we have for errors. Right now it's mostly about avoiding common false positives